packages/app/src/app/pages/Dashboard/Content/routes/Repositories/index.tsx

```diff
@@ -15,10 +15,34 @@ export const RepositoriesPage = () => {
     activeTeam,
     dashboard: { repositories },
   } = useAppState();
+  const pathRef = React.useRef<string>(null);
 
   React.useEffect(() => {
-    actions.dashboard.getRepositoriesByTeam();
-  }, [activeTeam, actions.dashboard]);
+    // If no repositories were fetched yet and the user tries
+    // to directly access a repository, we should fetch said
+    // repository only.
+    if (repositories === null) {
+      if (path) {
+        const [, owner, name] = path.split('/');
+        actions.dashboard.getRepositoryByDetails({ owner, name });
+      } else {
+        actions.dashboard.getRepositoriesByTeam();
+      }
+    }
+
+    // If the current view is the list of the repositories
+    // and the previous view was a repo and only that repo
+    // was fetched, get all repositories of that team.
+    if (
+      path === '' &&
+      pathRef.current?.startsWith('github') &&
+      repositories.length === 1
+    ) {
+      actions.dashboard.getRepositoriesByTeam();
+    }
+
+    pathRef.current = path;
+  }, [path]);
```

Comment:
nice! this ensures the repository page with all the branches doesn't load additional data when being open from the editor for example. one small note, maybe this component should have a computed field inside like `view: 'repositories' / 'branches'`

Reply:
@alexnm I must admit I spent more time on this than I should, at the of the day my reasoning was that the commentary and the single case were enough to explain the logic. But we can compute `currentPage` and `previousPage` if we want to handle more cases in this effect, like moving between repos (which is not possible atm).